test/e2e/backup_restore_test.go

```diff
@@ -0,0 +1,314 @@
+			if status != "completed" {
+				fmt.Println("Waiting for backup to be completed")
+			}
+			result = (string(body))
+			return status != "completed"
+		})
+	}
+
+	resp, err := client.R().
+		AddRetryCondition(retryFn).
+		SetHeaders(headerMap).
+		SetBodyString(dbReq).
+		Post(hostsEndpoint)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	return result, err
+}
+
+func TestMlDbBackupRestore(t *testing.T) {
+	// var resp *http.Response
+
+	var err error
+	var podName string
+	imageRepo, repoPres := os.LookupEnv("dockerRepository")
+	imageTag, tagPres := os.LookupEnv("dockerVersion")
+
+	if !repoPres {
+		imageRepo = "ml-docker-db-dev-tierpoint.bed-artifactory.bedford.progress.com/marklogic/marklogic-server-centos"
+		t.Logf("No imageRepo variable present, setting to default value: " + imageRepo)
+	}
+
+	if !tagPres {
+		imageTag = "11.1.0-centos-1.1.2"
+		t.Logf("No imageTag variable present, setting to default value: " + imageTag)
+	}
+
+	username := "admin"
+	password := "admin"
+
+	options := map[string]string{
+		"persistence.enabled":   "true",
+		"replicaCount":          "1",
+		"image.repository":      imageRepo,
+		"image.tag":             imageTag,
+		"auth.adminUsername":    username,
+		"auth.adminPassword":    password,
+		"logCollection.enabled": "false",
+	}
+
+	t.Logf("====Installing Helm Chart")
+	releaseName := "bkuprestore"
+
+	namespaceName := "ml-" + strings.ToLower(random.UniqueId())
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+
+	t.Logf("====Creating namespace: " + namespaceName)
+	k8s.CreateNamespace(t, kubectlOptions, namespaceName)
+
+	defer t.Logf("====Deleting namespace: " + namespaceName)
+	defer k8s.DeleteNamespace(t, kubectlOptions, namespaceName)
+
+	podName = testUtil.HelmInstall(t, options, releaseName, kubectlOptions)
+
+	t.Logf("====Describe pod for backup restore test")
+	k8s.RunKubectl(t, kubectlOptions, "describe", "pod", podName)
+
+	tlsConfig := tls.Config{}
+	// wait until the pod is in Ready status
+	k8s.WaitUntilPodAvailable(t, kubectlOptions, podName, 10, 15*time.Second)
+
+	tunnel7997 := k8s.NewTunnel(kubectlOptions, k8s.ResourceTypePod, podName, 7997, 7997)
+	defer tunnel7997.Close()
+	tunnel7997.ForwardPort(t)
+	endpoint7997 := fmt.Sprintf("http://%s", tunnel7997.Endpoint())
+
+	// verify if 7997 health check endpoint returns 200
+	http_helper.HttpGetWithRetryWithCustomValidation(
+		t,
+		endpoint7997,
+		&tlsConfig,
+		10,
+		15*time.Second,
+		func(statusCode int, body string) bool {
+			return statusCode == 200
+		},
+	)
+
+	//set test data path and documents to be loaded for tests
+	docPath := "../test_data/bkup_restore_reqs/"
+	docName := "testOne.xml"
+	docNameTwo := "testTwo.xml"
+
+	tunnel8000 := k8s.NewTunnel(kubectlOptions, k8s.ResourceTypePod, podName, 8000, 8000)
+	defer tunnel8000.Close()
+	tunnel8000.ForwardPort(t)
+	qConsoleEndpoint := fmt.Sprintf("http://%s/v1/documents?database=Documents&uri=%s", tunnel8000.Endpoint(), docName)
```

Comment:
Is it possible to insert or create both documents "docName" and "docNameTwo" as part of single request into ML DB?

Reply:
MarkLogic REST API: PUT /v1/documents allows only one uri to create or update a document.

Reply:
Can't we use POST https://docs.marklogic.com/REST/POST/v1/documents ?

Reply:
To use a single POST API call for creating multiple documents, we need to generate a multipart HTTP request body which can be generated using a Xquery and that would need a call to v1/eval API and, it's not recommended to use eval in tests. Please refer https://docs.marklogic.com/guide/rest-dev/bulk#id_85775